pkg/mesh/types.go

```diff
@@ -0,0 +1,31 @@
+package mesh
+
+import (
+	"context"
+
+	"github.com/deislabs/smi-sdk-go/pkg/apis/metrics"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Mesh interface {
+	// Return the list of supported resource kinds
+	GetSupportedResources(ctx context.Context) (*metav1.APIResourceList, error)
+	// Return metrics for a resource or for the type if name is empty
+	GetResourceMetrics(ctx context.Context,
+		name, namespace, kind string,
+		interval *metrics.Interval) (*metrics.TrafficMetricsList, error)
+	// Return the Edge Metrics for a resource
+	GetEdgeMetrics(ctx context.Context,
+		name, namespace, kind string,
+		interval *metrics.Interval,
+		details *ResourceDetails) (*metrics.TrafficMetricsList, error)
+}
+
+type Queries struct {
```

Comment:
Feels a little weird to have this here, I’d assume that different providers will use different structures here.